pkg/specgen/namespaces.go

```diff
@@ -238,19 +238,13 @@ func ParseNamespace(ns string) (Namespace, error) {
 	case ns == "private", ns == "":
 		toReturn.NSMode = Private
 	case strings.HasPrefix(ns, "ns:"):
-		split := strings.SplitN(ns, ":", 2)
-		if len(split) != 2 {
-			return toReturn, fmt.Errorf("must provide a path to a namespace when specifying \"ns:\"")
-		}
+		_, value, _ := strings.Cut(ns, ":")
```

Comment:
Wouldn't this be cleaner with `CutPrefix()`? Same with `container:` below, and then `auto` and `keep-id` and possibly more.

I'm very close to the end of my day, and this is as far as I'm likely to get.

This is really, really nice work. You've put a lot of effort into each individual replacement.

Reply:
CutPrefix is hard because we need to support both with & without `:`. Could do CutPrefix("auto") followed by CutPrefix(":")

Reply:
>we need to support both with & without :.

I was referring to line 309/303 below, `HasPrefix(ns, "auto:"):`. Likewise with the other `HasPrefix()` cases within a stone's throw of these few lines.

>might take a couple days to address feedback

Thank you for letting us know. That's perfect because it lets me (and hopefully others) review carefully without freaking out about time pressure.

Thank you again! This is a challenging review but I'm actually enjoying it.

Reply:
Converted the `switch {` to `switch ns {` in `ParseNamespace` & `ParseUserNamespace`. Moved prefix checks into default with `if/else if/else` chain

Reply:
I guess I was thinking of something much simpler: keeping the `switch` as-is, and:
```go
        ...
	case strings.HasPrefix(ns, "ns:"):
		value, _ := strings.CutPrefix(ns, "ns:")
		toReturn.NSMode = Path
		toReturn.Value = value
	case strings.HasPrefix(ns, "container:"):
		value, _ := strings.CutPrefix(ns, "container:")
		toReturn.NSMode = FromContainer
		toReturn.Value = value
```
To me that preserves the readability of the `switch` list (no complicated `switch` + `ifelse`) and also minimizes the diffs. Likewise in `ParseUserNamespace()` below. But maybe there's something I've missed?

Reply:
I see, bit annoying having repeated string constant, & it doubles the check, tho in theory compiler could realize that. I can change it to that if you'd prefer

Reply:
I agree that duplication is evil, but to my aesthetic sense it's the cleanest one to maintain. I'm not go-literate enough to have the final say, so I'll defer to my teammates.

Reply:
I am fine with this change as is, I think not having to parse the same thing twice is better.

Reply:
Thanks. @serprex when convenient could you please squash your commits and do one last rebase? I will monitor CI and try to approve in a timely manner.